src/redux/messageActions.js

```diff
@@ -35,11 +35,23 @@ export const updateChannelSize = newChannelSize => ({
 
 // Post Messages to Backend Thunk
 export const postMessagesThunk = () => (dispatch, getState) => {
-  return request("polls", 'POST', getState().message)
-    .then(() => {
-      console.log('data has been sent to back end')
-    })
-    .catch(error => {
-      console.log(error);
-    })
-};
\ No newline at end of file
+  const originalMessage = getState().message;
+  const clonedMessage = { ...originalMessage };
```

Comment:
why do you need to clone it?

Reply:
Hi Mustafa, to answer this plus some of the questions above,
We originally made our front end so that the responses in the state is an array. But on we realized on Thursday we had to make sure the responses is an object. I tried first converting everything related to responses to an object, but later on it wouldn't work to keep it in this structure since you can't really change the property names - and that's what's happening when the user is inputting a response (a way to do it is to delete the whole property and create a new one everytime the user types something, which makes the forms jump all over). So I had to revert back and keep it in the initial array, and then copy the state (can't manipulate it straight) and convert the responses to object, before sending it to the backend.

Reply:
Ok... but why does the backend want it as an object?

Reply:
Because it the properties will be the responses and the value are the votes (need a key value pair so we can have the votes per answer)